### DISCUSSION TITLE
Comparison to existing OSGi plugin

### DISCUSSION THREAD

**Discussion by lppedd** [JaneWardSandy/eclipse-pde-partial-idea#10] [General]
Hi! Currently I'm trying to move a bit of my RCP development from Eclipse to IDEA.
IDEA has a [plugin for OSGi development](https://plugins.jetbrains.com/plugin/1816-osgi), how does this one compare to that?

Btw, good work. RCP should really be a first-class citizen in IntelliJ.

**Comment by JaneWardSandy** [comment]
Hi, thank you for using this plugin.

I tried the Osmorc plugin before, and it has stopped adding new features for nearly 10 years after it was donate to JetBrains.

Osmorc support many OSGI implementation, including eclipse equinox. But for compatibility and flexibility, Eclipse equinox support is not very well, such as starting RCP application, dependency required and re-export prompts, kotlin support, etc.

So I create this plugin and try to use it resolve that problem, in the hope that I will be more smooth in the code writing process. Until now it, it is still in a very very early stage, and many features are not been implemented. 

The higher priority is let this plugin support us to debug and code writing, and other auxiliary functions are not planned. Some file such as plugins.xml and extension points still need to be written by manually or Eclipse.


Comparision:

Feature | PDE Partial | Osmorc
------- | ----------- | ------
Framework support | equinox Only | equinox, felix, concierge, knopflerfish
Project import support | x | by bnd
Bundle source | eclipse target | bundle directory
Manifest lint | partial support | partial support
Manifest auto gen | x | by bnd
Manifest manually edit | support with hint, reference navigate | support
Code inspection | support | support
Package accessibility | java, kotlin, import package, required bundle, re-export required bundle | java, import package only
Quick fix | import package, required bundle, required bundle with version | import package
Bundle startup manager | startup level only | startup level, auto startup
BND | x | by bnd
Maven | x | by bnd
Library | auto create | x
Facet | for binary package | for manifest file edit mode
Artifact | auto create for package | x
OSGI launche | x | support
OSGI console | x | support
Launche RCP | local, remote | x

**Comment by lppedd** [comment]
Thanks for the detailed comparison.
Maybe I'll give some help in the future.
